src/interfaces/IMetaMorpho.sol

```diff
@@ -70,36 +70,49 @@ interface IMetaMorphoBase {
-    /// @dev May be a little off `totalAssets()` after each interaction, due to some roundings.
+    /// @dev May be greater than `totalAssets()` due to removal of markets with non-zero supply or socialized bad debt.
+    /// This difference will decrease the fee accrued until one of the functions updating `lastTotalAssets` is
+    /// triggered (deposit/mint/withdraw/redeem/setFee/setFeeRecipient).
     function lastTotalAssets() external view returns (uint256);
 
     /// @notice Submits a `newTimelock`.
+    /// @dev Warning: Reverts if a timelock is already pending. Revoke the pending timelock to overwrite it.
     /// @dev In case the new timelock is higher than the current one, the timelock is set immediately.
-    /// @dev Warning: Submitting a timelock will overwrite the current pending timelock.
     function submitTimelock(uint256 newTimelock) external;
 
     /// @notice Accepts the pending timelock.
     function acceptTimelock() external;
 
     /// @notice Revokes the pending timelock.
+    /// @dev Does not revert if there is no pending timelock.
     function revokePendingTimelock() external;
 
     /// @notice Submits a `newSupplyCap` for the market defined by `marketParams`.
+    /// @dev Warning: Reverts if a cap is already pending. Revoke the pending cap to overwrite it.
+    /// @dev Warning: Reverts if a market removal is pending.
     /// @dev In case the new cap is lower than the current one, the cap is set immediately.
-    /// @dev Warning: Submitting a cap will overwrite the current pending cap.
     function submitCap(MarketParams memory marketParams, uint256 newSupplyCap) external;
 
-    /// @notice Accepts the pending cap of the market defined by `id`.
-    function acceptCap(Id id) external;
+    /// @notice Accepts the pending cap of the market defined by `marketParams`.
+    function acceptCap(MarketParams memory marketParams) external;
 
     /// @notice Revokes the pending cap of the market defined by `id`.
+    /// @dev Does not revert if there is no pending cap.
     function revokePendingCap(Id id) external;
 
-    /// @notice Submits a forced market removal from the vault, potentially losing all funds supplied to the market.
-    /// @dev Warning: Submitting a forced removal will overwrite the timestamp at which the market will be removable.
-    function submitMarketRemoval(Id id) external;
+    /// @notice Submits a forced market removal from the vault, eventually losing all funds supplied to the market.
+    /// @notice Funds can be recovered by enabling this market again and withdrawing from it (using `reallocate`),
+    /// but funds will be distributed pro-rata to the shares at the time of withdrawal, not at the time of removal.
+    /// @notice This forced removal is expected to be used as an emergency process in case a market constantly reverts.
+    /// To softly remove a sane market, the curator role is expected to bundle a reallocation that empties the market
```

Comment:
Consider also suggesting setting the cap to `0` to prevent any additional supply to a market that needs to be removed via the "normal" way. The market in this case can be removed via `updateWithdrawQueue` only the supply balance is equal to zero.

Reply:
Probably it's useless because you can be front-runned by a donation on behalf of MM. But it's still a good practice, I think

Reply:
If `submitMarketRemoval` this is required anyway to set the cap to 0 so I think it's not worth adding this comment

Reply:
As I suggested in another comment, I think that the best option will be to do a separate guide to list all the these "emergency" scenarios and what actions you need to perform to handle them. Also, having a utility contract that batch them (and have conditions inside based on the state of the pending proposals) will be nice to have.

Reply:
Just created an issue for that #375